src/groupeddataframe/groupeddataframe.jl

```diff
@@ -323,6 +323,14 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function Base.to_index(gd::GroupedDataFrame, key::Tuple{Vararg{<:Pair}})
+    println("hey there")
+    if length(key) != length(gd.cols) || any(n != _names(gd)[c] for (n, c) in zip(Symbol.(first.(key)), gd.cols))
```

Comment:
This seems to be minimally faster (but your implementation is also OK, so please leave what you prefer)
```suggestion
    if length(key) != length(gd.cols) || any(Symbol(first(k)) != _names(gd)[c] for (n, c) in zip(key, gd.cols))
```